File: 20210518-java-fantasybattle-refactoring/src/test/java/codingdojo/PlayerTest.java
package codingdojo;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import java.util.Collections;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.mock;

public class PlayerTest {

    // choose this one if you are familiar with mocks
    @Disabled("Test is not finished yet")
    @Test
    void damageCalculationsWithMocks() {
        Inventory inventory = mock(Inventory.class);
        Stats stats = mock(Stats.class);
        SimpleEnemy target = mock(SimpleEnemy.class);

        Damage damage = new Player(inventory, stats).calculateDamage(target);
        assertEquals(10, damage.getAmount());
    }

    // choose this one if you are not familiar with mocks
    @Test
    void damageCalculations() {
        BasicItem leftHand = new BasicItem("sword", 10, 1.0F);
        BasicItem rightHand = new BasicItem("shield", 0, 1.4F);
        BasicItem head = new BasicItem("helmet", 0, 1.2F);
        BasicItem feet = new BasicItem("boots", 0, 0.1F);
        BasicItem chest = new BasicItem("breastplate", 0, 1.4F);
        Equipment equipment = new Equipment(leftHand, rightHand, head, feet, chest);
        Inventory inventory = new Inventory(equipment);
        Stats stats = new Stats(0);
        SimpleEnemy target = new SimpleEnemy(new SimpleArmor(5), Collections.emptyList());

        Damage damage = new Player(inventory, stats).calculateDamage(target);

        assertEquals(46, damage.getAmount());
    }
}

File: 20210524-kotlin-fantasybattle-refactoring/src/test/kotlin/codingdojo/PlayerTest.kt
package codingdojo

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import org.junit.Test
import kotlin.test.assertEquals

class PlayerTest {

    @Test
    fun damageCalculationsWithMocks() {
        val inventory: Inventory = mock()
        val stats: Stats = mock()
        val target: SimpleEnemy = mock()
        whenever(target.armor).thenReturn(SimpleArmor(5))

        val leftHand = BasicItem("sword", 10, 1.0F)
        val rightHand = BasicItem("shield", 0, 1.4F)
        val head = BasicItem("helmet", 0, 1.2F)
        val feet = BasicItem("boots", 0, 0.1F)
        val chest = BasicItem("breastplate", 0, 1.4F)
        val equipment = Equipment(leftHand, rightHand, head, feet, chest)
        whenever(inventory.equipment).thenReturn(equipment)

        val damage = Player(inventory, stats).calculateDamage(target)

        assertEquals(50, damage.amount)
    }
}
